File: app/Models/Organization.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Organization extends Model
{
    protected $primaryKey = 'id';

    protected $keyType = 'int';

    public $incrementing = false;

    public function repositories()
    {
        return $this->hasMany(Repository::class, 'organization_id', 'id')
            ->orderBy('last_updated', 'desc');
    }

    public $fillable = [
        'id',
        'name',
        'description',
        'avatar_url',
    ];
}

File: app/models/organization.rb
class Organization < ApplicationRecord
  has_many :repositories
  has_many :public_repositories, -> { where(private: false) }, class_name: "Repository"

  def visible_repositories(include_private:)
    include_private ? repositories : public_repositories
  end

  def slug
    name.downcase.gsub(" ", "-")
  end

  def display_description
    if description.present?
      description
    else
      "No description provided."
    end
  end
end
